connector/servicegraphconnector/connector.go

```diff
@@ -593,7 +593,7 @@ func (p *serviceGraphConnector) buildMetricKey(clientName, serverName, connectio
 	var metricKey strings.Builder
 	metricKey.WriteString(clientName + metricKeySeparator + serverName + metricKeySeparator + connectionType)
 
-	for _, dimName := range p.config.Dimensions {
+	for _, dimName := range p.config.EdgeDimensions {
```

Comment:
Help me understand this small change - what will happen now to dimensions defined under `Dimensions`? Don't we need to check them here too anymore?

Reply:
In the buildMetricKey method, the edgeDimensions parameter is already prefixed, so Dimensions is not appropriate.

![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/25188468/f10d37fe-92c5-44fb-8401-8360beab3dfe)

If you had added `application.key` to Demensions, edgeDimensions would only have `client_application.key` and `server_application.key`, but not application.key. For this reason, you don't need to check Demensions.

Reply:
But `edgeDimensions` are not prefixed because they're the plain span attributes, no? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/servicegraphconnector/connector.go#L255

Reply:
Yes, you're right, you need to create a metricKey with dimensions prefixed so that you can manage your metrics correctly.

As I keep saying, if you try to create a metric key using app.key in Dimensions, it will be prefixed with client_app.key, server_app.key, so you can't set it normally when creating a metric key. That's why you need EdgeDimensions

Is the purpose of Dimensions just for prefixing?

Reply:
I see.

So I want to create a metricKey using the prefixed client_app.key and server_app.key, but do I understand correctly that this is not necessary?

Reply:
One last question: how do you aggregate prefixed metrics?

Reply:
I'm not sure I understand what the problem is, sorry.

>One last question: how do you aggregate prefixed metrics?

I guess that depends on the metric's product/project that you use.